### DISCUSSION TITLE
How one can process the source code in a one file structure?

### DISCUSSION THREAD

**Discussion by Davoodeh** [aristocratos/bpytop#285] [Q&A] [answered]
I'm currently obsessed with this tool and wanted to use the style and general feeling in my own scripts and improve this one.

As a newbie, I have a hard time reading the file. Is there a tool used to tailor different parts of this one file together?
Or the author just went head first in this 5k+ lines script?
Are there any tools that can help me have an easier time around it? Any documentation on how the code is structured?

On a more general side: Is it better to use this as a template or something like "Prompt toolkit"? (any other noob-friendly alternative to it?)

I'd be more than glad if one could provide any helpful resources.
P.S: The wiki is empty.
(Sorry if this was a wrong place for posting this)

**Comment by aristocratos** [comment, marked as answer]
@Davoodeh 
Well, in general I would recommend vscode to get a better overview when coding regardless of language or how big the file is. There is loads of useful plugins and it provides a really nice outline to make your life easier and quickly jump around big files. This is my coding setup for bpytop: 
![vscode](https://user-images.githubusercontent.com/59659483/112665039-ea0f3c80-8e5a-11eb-9c24-33d282218560.png)

However using bpytop as a kinda framework for writing other programs is not gonna be easy unless you have got a good grasp of classes, subclasses and class methods. All core functions; text buffering, keyboard/mouse input, and etc., was written specifically for use in bpytop so might need a lot of modifying to work well in other projects.

The only documentation currently existing is the sparse comments I've left in the code.
The names of functions, classes and variables (except some integers) should be fairly descriptive of their intended purpose.
And I think I've left comments for the bit more confusing parts of the code, otherwise the code should be fairly self explanatory if you've got some experience with python.

There is some great documentation for psutil (the library bpytop is heavily dependent on) at: https://psutil.readthedocs.io/en/latest/
All other libraries used are standard libraries and should have good documentation at: https://docs.python.org/3/

If you have questions about specific parts of the code feel free to open new discussion threads and I will try to answer when I've got time :)

**Reply by Davoodeh** [reply]
Thank you sincerely
